### PR TITLE
Document CAS auth

### PR DESCRIPTION
**Pull request by stevenhammerton**
Quick documentation of CAS auth.

### CODE REVIEW

File: specification/client_server_api.rst [hunk 1]
```diff
@@ -290,6 +290,33 @@ the auth code. Home servers can choose any path for the ``redirect URI``. Once
...
+  Authentication is supported via CAS URLs. This login is based on redirection of
+  the user to multiple endpoints.
+:Parameters:
+  ``redirectUrl``: Client URL to redirect to after the homeserver has authenticated
+  the user with CAS.
+
+CAS login works by redirecting the user to the CAS server endpoint via the homeserver. 
+
+The login process is started by a client redirecting a user to the endpoint
+/login/cas/redirect on the homeserver client API. When redirecting to this endpoint
```
Comment (Kegsay) [inline specification/client_server_api.rst]: Backticks on the HTTP path please :3

File: specification/client_server_api.rst [hunk 2]
```diff
@@ -290,6 +290,33 @@ the auth code. Home servers can choose any path for the ``redirect URI``. Once
...
+the client should include a redirectUrl query parameter to which the user will be
```
Comment (Kegsay) [inline specification/client_server_api.rst]: Backticks on `redirectUrl` please.

File: specification/client_server_api.rst [hunk 3]
```diff
@@ -290,6 +290,33 @@ the auth code. Home servers can choose any path for the ``redirect URI``. Once
...
+redirected after the homeserver has authenticated the user with CAS.
+
+The homeserver then redirects the user to CAS which authenticates the user and redirects
+the user back to the homeserver with a one time ticket. The homeserver can use the ticket
+to validate the authentication with CAS.
+
+If successful, the user is redirected back to the client via the redirectUrl given to the
+homeserver on the initial redirect request. In the url will be a loginToken query parameter
```
Comment (Kegsay) [inline specification/client_server_api.rst]: Backticks on `loginToken` please.

File: specification/client_server_api.rst [hunk 4]
```diff
@@ -290,6 +290,33 @@ the auth code. Home servers can choose any path for the ``redirect URI``. Once
...
+The login process is started by a client redirecting a user to the endpoint
+/login/cas/redirect on the homeserver client API. When redirecting to this endpoint
+the client should include a redirectUrl query parameter to which the user will be
+redirected after the homeserver has authenticated the user with CAS.
+
+The homeserver then redirects the user to CAS which authenticates the user and redirects
+the user back to the homeserver with a one time ticket. The homeserver can use the ticket
+to validate the authentication with CAS.
+
+If successful, the user is redirected back to the client via the redirectUrl given to the
```
Comment (Kegsay) [inline specification/client_server_api.rst]: And what if it is not successful?

**Comment by Kegsay** [comment]
LGTM other than comments.

**Comment by richvdh** [comment]
@stevenhammerton: would you have a chance to address kegan's comments? It would be great to get this merged.

**Comment by ara4n** [comment]
@stevenhammerton any chance you can feedback on @kegsay's stuff so we can merge this? :)

**Comment by Kegsay** [comment]
I would fix the things I brought up myself but I'm not sure what happens in the event of a failure for https://github.com/matrix-org/matrix-doc/pull/171#discussion_r45072844

**Comment by richvdh** [comment]
closing in favour of #367

**Pull request closed without merging**